### PR TITLE
Remove dbus-c++ dependencies of examples

### PR DESCRIPTION
**Pull request by kursatkobya**
Signed-off-by: kursat <kursat.kobya@pelagicore.com>

### CODE REVIEW

File: examples/temperature/temperature-service-console-client/temperatureserviceinterface.cpp [hunk 1]
```diff
@@ -22,42 +22,50 @@
...
-    std::ofstream logfile(m_logfilepath.c_str(), std::ofstream::app);
-    if(logfile.is_open()) {
-        logfile << __TIMESTAMP__ << " - Temperature changed to: "
-                << temperature << "°C" << std::endl;
-        logfile.close();
-    } else {
-        std::cerr << m_logfilepath << " could not be opened to write." << std::endl;
-    }
+    m_proxy->SetTemperature(temperature,
+                            sigc::mem_fun(*this,
```
Comment (DunderRoffe) [inline examples/temperature/temperature-service-console-client/temperatureserviceinterface.cpp]: Would be nice to extract the `mem_fun` stuff to `auto callback = sigc::mem_fun...`

File: examples/temperature/temperature-service/src/CMakeLists.txt [hunk 2]
```diff
@@ -15,18 +15,16 @@
 #
 # For further information see LICENSE
```
Comment (DunderRoffe) [inline examples/temperature/temperature-service/src/CMakeLists.txt:17]: Would be nice to have minimum cmake version here

File: examples/temperature/temperature-service/src/temperatureservice_dbus_common.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    void ret()
+    {
+        std::vector<Glib::VariantBase> vlist;
+
+        m_message->return_value(Glib::Variant<Glib::VariantBase>::create_tuple(vlist));
+    }
+
+    //template <typename T0>
```
Comment (breakreturn) [inline examples/temperature/temperature-service/src/temperatureservice_dbus_common.h]: Any reason to keep this and the similar comment below?

**Review by breakreturn — CHANGES_REQUESTED** [review]
Just a minor comment

File: examples/temperature/temperature-service-console-client/temperatureserviceinterface.cpp [hunk 4]
```diff
@@ -22,42 +22,50 @@
...
+    double temperature;
+    m_proxy->GetTemperature_finish(temperature, result);
+    std::cout << "Temperature : " << temperature << std::endl;
+}
+
+void TemperatureServiceInterface::getTemperature()
 {
-    SetTemperature(temperature);
+    m_proxy->GetTemperature(sigc::mem_fun(*this,
+                                          &TemperatureServiceInterface::on_gettemp_finished));
```
Comment (DunderRoffe) [inline examples/temperature/temperature-service-console-client/temperatureserviceinterface.cpp]: Would be nice to extract the `mem_fun` stuff to `auto callback = sigc::mem_fun...`

**Review by DunderRoffe — CHANGES_REQUESTED** [review]
Looks really nice overall but a few things:

File: examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ * For further information see LICENSE
+ */
+
+
+#pragma once
+#include <string>
+#include <glibmm.h>
+#include <giomm.h>
+#include "../temperature-service/src/temperatureservice_dbus_common.h"
```
Comment (tobsan) [inline examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h]: Do you really need this one? Can't you generate this for the client separately?

**Review by tobsan — CHANGES_REQUESTED** [review]
Looks generally good code-wise, but the client no longer logs to a file from what I can see. Why not?

File: examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h (hunk 5 above)
Comment (kursatkobya) [reply examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h]: i can put same file to client as well but then we had redundant files.

File: examples/temperature/temperature-service/src/CMakeLists.txt (hunk 2 above)
Comment (kursatkobya) [reply examples/temperature/temperature-service/src/CMakeLists.txt:17]: cmake version is applicable with SC (3.0.2)

File: examples/temperature/temperature-service/src/temperatureservice_dbus_common.h (hunk 3 above)
Comment (DunderRoffe) [reply examples/temperature/temperature-service/src/temperatureservice_dbus_common.h]: No this should be removed. It is generated  :/

File: examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h (hunk 5 above)
Comment (kursatkobya) [reply examples/temperature/temperature-service-console-client/temperature_dbus_proxy.h]: But after a trial, this include can be removed totally.

**Review by DunderRoffe — APPROVED** [review]

File: examples/temperature/temperature-service/src/CMakeLists.txt (hunk 2 above)
Comment (DunderRoffe) [reply examples/temperature/temperature-service/src/CMakeLists.txt:17]: Actually it is already present :) nwm

**Review by breakreturn — APPROVED** [review]

**Comment by tobsan** [comment]
I still don't see this logging to file anywhere. Either fix that, or change the script that runs this example so that a log file is not needed anymore.

**Review by tobsan — APPROVED** [review]

**Pull request merged**